src/huggingface_hub/utils/endpoint_helpers.py

```diff
@@ -14,10 +14,55 @@
 with the aim for a user-friendly interface
 """
 
+import math
+import re
 from dataclasses import dataclass
 from typing import List, Union
 
 
+def _filter_emissions(
+    models,
+    minimum_threshold: Union[int, float] = None,
+    maximum_threshold: Union[int, float] = None,
+):
+    """Filters a list of models for those that include an emission tag
+    and limit them to between two thresholds
+
+    Args:
+        models (:obj:`ModelInfo` or :class:`List`):
+            A list of `ModelInfo`'s to filter by
+        minimum_threshold (:obj:`int` or :obj:`float`):
+            A minimum carbon threshold to filter by, such as 1.
+        maximum_threshold (:obj:`int` or :obj:`float):
+            A maximum carbon threshold to filter by, such as 10.
+    """
+    if minimum_threshold is None and maximum_threshold is None:
+        raise ValueError(
+            "Both `minimum_threshold` and `maximum_threshold` cannot both be `None`"
+        )
+    if not minimum_threshold:
```

Comment:
(nit) this will be `False` if `minimum_treshold` is set to `0`, should it be  `if minimum_treshold is not None` ?

Reply:
Makes sense. Went with
```
if minimum_threshold is None:
```
(As this sets `minimum_threshold` to be a default value)